docs/how-to-guides/debug-builds.rst

```diff
@@ -0,0 +1,49 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _how-to-guides-debug-builds:
+
+******************
+Using debug builds
+******************
+
+For development work on native code in your Python package, you may want to use
+a debug build. To do so, we need to use the ``buildtype`` argument, which is
+equivalent to ``-Ddebug=true -Doptimization=0``, to ``meson setup``. In addition,
+it is likely most useful to set up an editable build with a fixed build
+directory. That way, the shared libraries in the installed debug build will
+contain correct paths, rather than paths pointing to a temporary build
+directory which ``meson-python`` will otherwise use. IDEs and other tools will
+then be able to show correct file locations and line numbers during debugging.
+
+We can do all that with the following ``pip`` invocation:
+
+.. code-block:: bash
```

Comment:
Using `.. code-block:: console` for command line examples formats them in a slightly nicer way.

Reply:
adopted, thanks